crates/utils/src/attr/attributes.rs

```diff
@@ -59,15 +64,40 @@ where
     }
 }
 
-impl<Bool, Num> Default for Attributes<Bool, Num>
+impl<Bool, Num, Float> AttrCtx<Float, f64> for Attributes<Bool, Num, Float>
 where
     Bool: FromStr + Hash + Eq + Copy,
     Num: FromStr + Hash + Eq + Copy,
+    Float: FromStr + Hash + Eq + Copy,
+{
+    fn get(&self, attr: Float) -> Option<&f64> {
+        self.float_attrs.get(attr)
+    }
+
+    fn get_loc(&self, attr: Float) -> Option<GPosIdx> {
+        self.float_attrs.get_loc(attr)
+    }
+
+    fn set(&mut self, attr: Float, value: f64, loc: GPosIdx) {
+        self.float_attrs.set(attr, value, loc);
+    }
+
+    fn remove(&mut self, attr: Float) {
+        self.float_attrs.remove(attr);
+    }
+}
+
+impl<Bool, Num, Float> Default for Attributes<Bool, Num, Float>
+where
+    Bool: FromStr + Hash + Eq + Copy,
+    Num: FromStr + Hash + Eq + Copy,
+    Float: FromStr + Hash + Eq + Copy,
 {
     fn default() -> Self {
         Self {
             num_attrs: AttrStore::default(),
             bool_attrs: AttrStore::default(),
+            float_attrs: AttrStore::default(),
```

Comment:
If you were really hankering for performance you could probably do some shenanigans to store floats and ints in the same array and only use 31-bits for the information and 1 bit for tracking the datatype

Reply:
I don't think this matters in the long term especially if we want string attributes someday, mainly worried about code duplication right now lol, but macro-izing everything seems like a pain.

Reply:
Yeah, just an idle thought. String attributes would probably live on the heap behind a Box. But yeah, this looks fine.